src/main/java/helper/RequestValidator.java

```diff
@@ -12,25 +13,25 @@
 
 	private List<String> getMissingFields(Map<String, Object> data, String[] required) {
```

Comment:
Ideally this should return ImmutableList instead of the callers of this making a copy.

FYI, you can use ImmutableList.Builder

```
ImmutableList.Builder<String> missing = ImmutableList.builder();

for (String s : required) {
  if (!data.containsKey(s)) {
    missing.add(s);
  }
}

return missing.build();
```

Alternatively, you can also use the streams and collectors.

```
import static com.google.common.collect.ImmutableList.toImmutableList;

return required.stream()
  .filter(s -> !data.containsKey(s))
  .collect(toImmutableList());
```

I think the latter is cleaner.

Reply:
Done.